Docs: Final edits, Style enhancements, Vale checks

sheilamjones opened the pull request:
This PR fixes style issues (title capitalization, one sentence per line...), links, Vale issues, and some grammar as per the preferred Quarkus Contributor guidelines in the security-oidc-code-flow-authentication.adoc (OIDC code flow authentication guide)

Reference: QDOCS-537

sheilamjones committed:
QDOCS-537: edits, vale checks post 3.2

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -85,23 +86,25 @@ To work around this limitation, you can configure the missing endpoint URL value
-# Configure the end-session URL locally, it can be an absolute or relative (to 'quarkus.oidc.auth-server-url') address
-quarkus.oidc.end-session-path=logout
+# Configure the end-session URL locally.
+It can be an absolute or relative (to 'quarkus.oidc.auth-server-url') address
```
```suggestion
# It can be an absolute or relative (to 'quarkus.oidc.auth-server-url') address
```

It is a comment in the properties file

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -85,23 +86,25 @@ To work around this limitation, you can configure the missing endpoint URL value
-quarkus.oidc.end-session-path=logout
+# Configure the end-session URL locally.
+It can be an absolute or relative (to 'quarkus.oidc.auth-server-url') address
+quarkus.oidc.end-session-path=logout.
```
```suggestion
quarkus.oidc.end-session-path=logout
```

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -161,12 +164,12 @@ quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
 
 # This is a key which will be used to retrieve a secret from the map of credentials returned from CredentialsProvider
-quarkus.oidc.credentials.jwt.secret-provider.key=mysecret-key
+quarkus.oidc.credentials.jwt.secret-provider.key=mysecret-key.
```
```suggestion
quarkus.oidc.credentials.jwt.secret-provider.key=mysecret-key
```

sheilamjones replied:
Removed period again per your suggestion. Thanks for clarifying this.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -161,12 +164,12 @@ quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
 ----
 
-Example of `private_key_jwt` with the PEM key file, and where the signature algorithm is RS256:
+Example of `private_key_jwt` with the Privacy Enhanced Mail (PEM) key file, and where the signature algorithm is RS256:
```
```suggestion
Example of `private_key_jwt` with the PEM key file, and where the signature algorithm is RS256:
```

We don't really expand PEM like that, it would probably confuse some users even if it is correct

sheilamjones replied:
Reverted this change per your suggestion.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -203,27 +207,27 @@ quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
 quarkus.oidc.credentials.jwt.audience=${quarkus.oidc-client.auth-server-url}
 
-# custom subject instead of the client id :
+# custom subject instead of the client ID:
```
```suggestion
# custom subject instead of the client id
```
We have discussed it with Rolfe, I use capital `ID` only when talking about ID tokens.
Also, this is not a real doc but a comment in the properties, so here `:` looks unusual

sberyozkin replied:
Lets keep `:` in some comments, sure, updating my suggestions to keep them as you intended

sheilamjones replied:
Thanks for clarifying @sberyozkin. I reverted these instances of "ID" back to "id". I do think though that as we are talking about "client ID" in generic terms here and not the specific property name, that "ID" should be capitalized. However, for consistency with your other discussions, I will revert back to "id" here and touch base also with Rolfe.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -203,27 +207,27 @@ quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
 quarkus.oidc.credentials.jwt.token-key-id=mykey
 
-# Use RS512 signature algorithm instead of the default RS256
+# Use RS512 signature algorithm instead of the default RS256:
```
```suggestion
# Use RS512 signature algorithm instead of the default RS256
```

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -203,27 +207,27 @@ quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
 quarkus.oidc.credentials.jwt.subject=custom-subject
 
-# custom issuer instead of the client id :
+# custom issuer instead of the client ID:
```
```suggestion
# custom issuer instead of the client id
```

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -329,7 +333,7 @@ public class OidcTokenRequestCustomizer implements OidcRequestFilter {
 <2> Use `OidcConfigurationMetadata` to filter requests to the OIDC token endpoint only.
 
-Alternatively, you can use `OidcRequestFilter.Endpoint` enum to make sure this filter is applied to the token endpoint requests only:
+Alternatively, you can use `OidcRequestFilter.Endpoint` enum to verify this filter is applied to the token endpoint requests only:
```
```suggestion
Alternatively, you can use `OidcRequestFilter.Endpoint` enum to have this filter applied to the token endpoint requests only:
```

`verify` does not work

sheilamjones replied:
Thanks @sberyozkin:
Propose an additional slight change to avoid the passive voice:
"Alternatively, you can use `OidcRequestFilter.Endpoint` enum to apply this filter to the token endpoint requests only:"

sheilamjones replied:
Updated to "Alternatively, you can use OidcRequestFilter.Endpoint enum to apply this filter to the token endpoint requests only:"

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -354,21 +358,22 @@ public class OidcDiscoveryRequestCustomizer implements OidcRequestFilter {
 }
 ----
-<1> Restrict this filter to requests targeting the OIDC discovery endpoint only.
+<1> Restrict this filter to request targeting of the OIDC discovery endpoint only.
```
```suggestion
```

sberyozkin replied:
IMHO it does not quite work

sheilamjones replied:
Thanks @sberyozkin. Just to confirm, is your suggestion to remove sentence 361 entirely?

sberyozkin replied:
@sheilamjones Please remove it

sheilamjones replied:
Done

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -522,48 +538,49 @@ This is done by ensuring tokens can be trusted.
-With Quarkus `web-app` applications, only the `IdToken` is verified by default because the access token is not used to access the current Quarkus `web-app` endpoint and is intended to be propagated to the services expecting this access token.
-If you expect the access token to contain the roles required to access the current Quarkus endpoint (`quarkus.oidc.roles.source=accesstoken`) then it will also be verified.
---
+===
```
Sometimes I'm seeing `====` with the Note, is `===` correct ?

sheilamjones replied:
Good catch. The correct formatting is "====" and not "===" in order for the Note to render correctly (and not display "===".

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -522,48 +538,49 @@ This is done by ensuring tokens can be trusted.
 ==== Further security with Proof Key for Code Exchange (PKCE)
 
-link:https://datatracker.ietf.org/doc/html/rfc7636[Proof Key for Code Exchange] (PKCE) minimizes the risk of authorization code interception.
+link:https://datatracker.ietf.org/doc/html/rfc7636[Proof Key for code exchange] (PKCE) minimizes the risk of authorization code interception.
```
```suggestion
link:https://datatracker.ietf.org/doc/html/rfc7636[Proof Key for Code Exchange] (PKCE) minimizes the risk of authorization code interception.
```

Lets keep it line with how RFC refers to it

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -608,11 +626,12 @@ OIDC `CodeAuthenticationMechanism` uses the default `io.quarkus.oidc.TokenStateM
+For example, storing tokens in the database or other server-side storage, if you prefer and have good reasons for storing the token state on the server.
 
-You can configure the default `TokenStateManager` to avoid saving an access token in the session cookie and only keep ID and refresh tokens or ID token only.
+You can configure the default `TokenStateManager` to avoid saving an access token in the session cookie and to only keep ID and refresh tokens or ID tokens only.
```
```suggestion
You can configure the default `TokenStateManager` to avoid saving an access token in the session cookie and to only keep ID and refresh tokens or ID token only.
```

session cookie can keep a single ID token

sheilamjones replied:
Thanks for clarifying. For clarity, I updated this to say 
"session cookie and to only keep ID and refresh tokens or a single ID token only."

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -789,10 +810,10 @@ implementation("io.quarkus:quarkus-reactive-pg-client")
 ----
 
-And you are ready to go.
+You are done.
```
Reads unusual, maybe:

```suggestion
Now the tokens will be stored in the database
```

sheilamjones replied:
Thanks for your suggestion, that is much better and provides more information. One small adjustment because  having "Now" and "will be" in the same sentence is slightly contradictory. What about:
"Now, the tokens are ready to be stored in the database."  
The next line 815 then provides more information: 
"By default, a database table used for storing tokens is created for you, however, you can disable this option with the `quarkus.oidc.db-token-state-manager.create-database-table-if-not-exists` configuration property."

@sberyozkin, would this work?

sheilamjones replied:
Thanks for your thumbs up @sberyozkin. Updated per suggestion.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -838,21 +859,21 @@ Let's start with explicit logout operations.
 Note that a cookie name varies when using xref:security-openid-connect-multitenancy.adoc[OpenID Connect Multi-Tenancy]. For example, it will be named `q_post_logout_tenant_1` for a tenant with a `tenant_1` ID, and so on.
 
-Here is an example of how to configure a Quarkus app initiated logout flow:
+Here is an example of how to configure a logout flow initiated by a Quarkus application:
```
```suggestion
Here is an example of how to configure Quarkus application to initiate a logout flow:
```

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -931,17 +954,17 @@ quarkus.oidc.logout.backchannel.path=/back-channel-logout
 This option is rarely used.
 
-You can configure Quarkus to support `Front-Channel Logout` as follows:
+You can configure Quarkus to support front-channel logout as follows:
```
```suggestion
You can configure Quarkus to support Front-channel logout as follows:
```

sheilamjones replied:
I also updated the earlier reference then from "back-channel logout" to "Back-channel logout" (line 942)

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1027,54 +1052,55 @@ This refresh token might also be recycled (refreshed) itself as part of this pro
 === Integration with GitHub and non-OIDC OAuth2 providers
 
-Some well known providers such as GitHub or LinkedIn are not OpenID Connect but OAuth2 providers which support the `authorization code flow`, for example, link:https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps[GitHub OAuth2] and link:https://docs.microsoft.com/en-us/linkedin/shared/authentication/authorization-code-flow[LinkedIn OAuth2].
+Some well-known providers such as GitHub or LinkedIn are not OpenID Connect provides, but OAuth2 providers that support the `authorization code flow`, for example, link:https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps[GitHub OAuth2] and link:https://docs.microsoft.com/en-us/linkedin/shared/authentication/authorization-code-flow[LinkedIn OAuth2].
```
```suggestion
Some well-known providers such as GitHub or LinkedIn are not OpenID Connect providers but OAuth2 providers that support the `authorization code flow`, for example, link:https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps[GitHub OAuth2] and link:https://docs.microsoft.com/en-us/linkedin/shared/authentication/authorization-code-flow[LinkedIn OAuth2].
```

sheilamjones replied:
Oops, my apologies for the typo. Good catch and fixed now.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1027,54 +1052,55 @@ This refresh token might also be recycled (refreshed) itself as part of this pro
---
-Even though you configure the extension to support the authorization code flows without `IdToken`, an internal `IdToken` is generated to standardize the way `quarkus-oidc` operates.
+====
+Eventhough you configure the extension to support the authorization code flows without `IdToken`, an internal `IdToken` is generated to standardize the way `quarkus-oidc` operates.
```
```suggestion
Even though you configure the extension to support the authorization code flows without `IdToken`, an internal `IdToken` is generated to standardize the way `quarkus-oidc` operates.
```

Have never seen it as a single word before

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1301,9 +1328,9 @@ public class CustomJavaScriptRequestChecker implements JavaScriptRequestChecker
 ----
 
-and reload the last requested page in case of a `499` status code.
+Reload the last requested page in case of a `499` status code.
```
It kind of breaks the flow of the example `and reload` is correct IMHO

sheilamjones replied:
Reverted these style changes back again.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1657,15 +1689,17 @@ public class CodeFlowAuthorizationTest {
-By default, `KeycloakTestResourceLifecycleManager` uses HTTPS to initialize a Keycloak instance which can be disabled with `keycloak.use.https=false`.
-Default realm name is `quarkus` and client id - `quarkus-web-app` - set `keycloak.realm` and `keycloak.web-app.client` system properties to customize the values if needed.
+By default, `KeycloakTestResourceLifecycleManager` uses HTTPS to initialize a Keycloak instance that can be disabled by specifying `keycloak.use.https=false`.
+The default realm name is `quarkus` and client ID is `quarkus-web-app` - set `keycloak.realm` and `keycloak.web-app.client` system properties to customize the values if needed.
```
```suggestion
The default realm name is `quarkus` and client id is `quarkus-web-app` - set `keycloak.realm` and `keycloak.web-app.client` system properties to customize the values if needed.
```

sheilamjones replied:
I will change this back for consistency with feedback elsewhere, however, I do think "ID" is correct here. Perhaps we can revisit this discussion again at a later point.

sberyozkin commented:
Thanks @sheilamjones for this quality update, a lot of changes, I've proposed a few updates but overall looks great

sheilamjones commented:
Many thanks @sberyozkin for your review. I have implemented your proposals and reverted some changes per your suggestions. I did have just a couple of outstanding questions.

sberyozkin commented:
Thanks @sheilamjones, your suggestions make sense.

I won't have time though to merge this PR today, I'll be signing off pretty soon, I believe you will also expect someone else from the Doc team to review, so I'll deal with finalizing this merge on 15 or 16th Jan,

Cheers

MichalMaler commented:
@sheilamjones 
Line 33 - Would this be better?
. The OIDC provider authenticates the user credentials entered and, if successful, issues an authorization code **_and_** redirects the user back to the Quarkus web-app with the code included as a query parameter.

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -52,11 +52,12 @@ For information about how to support multiple tenants, see xref:security-openid-
 === Configuring access to the OIDC provider endpoint
 
-The OIDC `web-app` application requires URLs of the OIDC provider's authorization, token, `JsonWebKey` (JWK) set, and possibly the `UserInfo`, introspection, and end session (RP-initiated logout) endpoints.
+The OIDC `web-app` application requires URLs of the OIDC provider's authorization, token, `JsonWebKey` (JWK) set, and possibly the `UserInfo`, introspection and end-session (RP-initiated logout) endpoints.
```
I will prepare a slight rework for this section, but it will be done in the following PR.

sheilamjones replied:
Hi @MichalMaler, did you have additional feedback for this section to add here?

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -329,7 +333,7 @@ public class OidcTokenRequestCustomizer implements OidcRequestFilter {
 <1> Get `OidcConfigurationMetadata` which contains all supported OIDC endpoint addresses.
```
```suggestion
<1> Get `OidcConfigurationMetadata`, which contains all supported OIDC endpoint addresses.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -329,7 +333,7 @@ public class OidcTokenRequestCustomizer implements OidcRequestFilter {
 <1> Get `OidcConfigurationMetadata` which contains all supported OIDC endpoint addresses.
 <2> Use `OidcConfigurationMetadata` to filter requests to the OIDC token endpoint only.
```
```suggestion
<2> Use `OidcConfigurationMetadata` to only filter requests to the OIDC token endpoint.
```

sheilamjones replied:
@MichalMaler, I think your proposed update changes the intended meaning here, that is, the original suggests that requests can be filtered to the "OIDC token endpoint" only (and not to any other type of endpoint), whereas, your changes state that OidcConfigurationMetadata can only "requests" (and not any other type of data) to the OIDC token endpoint.
@sberyozkin, which is the intended meaning here?

sheilamjones replied:
Thanks @sberyozkin. Shall keep it as it is then. Thank you.

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -386,11 +390,12 @@ quarkus.oidc.authentication.extra-params.response_mode=query
 
-In such cases, an HTTP `401` error will be returned by default.
+In such cases, an HTTP `401` error is returned by default.
 However, you can instead request that a custom public error endpoint is called to return a more user-friendly HTML error page.
```
```suggestion
However, you can request that a custom public error endpoint be called to return a more user-friendly HTML error page.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -400,22 +405,25 @@ quarkus.oidc.authentication.error-path=/error
-It is important to ensure that this error endpoint is a public resource to prevent the user from being redirected to this page to be authenticated again.
+[IMPORTANT]
+====
+To prevent the user from being redirected to this page to be reauthenticated, ensure that this error endpoint is a public resource.
```
```suggestion
Ensure that this error endpoint is a public resource to prevent the user from being redirected to this page to be re-authenticated.
```

sheilamjones replied:
@MichalMaler, the sequence according to IBM style should be "what" before "how". Therefore, I propose to keep this as is, that is, "To prevent......, ensure that....". Good catch though on the missing hyphen in "re-authenticated". Fixed this.

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -400,22 +405,25 @@ quarkus.oidc.authentication.error-path=/error
 === Accessing authorization data
 
-Let's first discuss how to access information around authorization.
+You can access information around authorization in different ways.
```
```suggestion
You can access information about authorization in different ways.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -400,22 +405,25 @@ quarkus.oidc.authentication.error-path=/error
 
 [[access_id_and_access_tokens]]
 ==== Accessing ID and access tokens
 //SJ: new concept topic to describe the different token types and usage proposed in next iteration
```
```suggestion
//SJ: New concept topic to describe the different token types and usage proposed in the next iteration.
```

sheilamjones replied:
Just a comment that I added to this topic. Shall leave this as is.

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -400,22 +405,25 @@ quarkus.oidc.authentication.error-path=/error
+The OIDC code authentication mechanism acquires three tokens during the authorization code flow: https://openid.net/specs/openid-connect-core-1_0.html#IDToken[ID token], access token, and refresh token.
 
-ID Token is always a JWT token and is used to represent a user authentication with the JWT claims.
+The ID token is always a JWT token and is used to represent a user authentication with the JWT claims.
```
```suggestion
The ID token is always a JWT token and represents a user authentication with the JWT claims.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -439,7 +447,8 @@ public class ProtectedResource {
 ----
 
-Access Token is usually used by the OIDC `web-app` application to access other endpoints on behalf of the currently logged-in user. The raw access token can be accessed as follows:
+The access token is usually used by the OIDC `web-app` application to access other endpoints on behalf of the currently logged-in user.
```
```suggestion
The OIDC `web-app` application usually uses the access token to access other endpoints on behalf of the currently logged-in user.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -473,45 +482,51 @@ public class ProtectedResource {
 [[user-info]]
 ==== User info
 
 If the ID token does not provide enough information about the currently-authenticated user, you can get more information from the `UserInfo` endpoint.
```
```suggestion
If the ID token does not provide enough information about the currently authenticated user, you can get more information from the `UserInfo` endpoint.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -473,45 +482,51 @@ public class ProtectedResource {
-Additionally, a custom `SecurityIdentityAugmentor` can also be used to add the roles. For more information, see xref:security-customization.adoc#security-identity-customization[SecurityIdentity customization].
+Additionally, you can also use a custom `SecurityIdentityAugmentor` to add the roles.
+For more information, see xref:security-customization.adoc#security-identity-customization[SecurityIdentity customization].
 You can also map `SecurityIdentity` roles created from token claims to deployment specific roles with the xref:security-authorize-web-endpoints-reference.adoc#map-security-identity-roles[HTTP Security policy].
```
```suggestion
You can also map `SecurityIdentity` roles created from token claims to deployment-specific roles with the xref:security-authorize-web-endpoints-reference.adoc#map-security-identity-roles[HTTP Security policy].
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -522,27 +537,28 @@ This is done by ensuring tokens can be trusted.
 
-For information about the claim verification, including the `iss` (issuer) claim, see the xref:security-oidc-bearer-token-authentication.adoc#jwt-claim-verification[JSON Web Token Claim verification] section.
+For information about the claim verification, including the `iss` (issuer) claim, see the xref:security-oidc-bearer-token-authentication.adoc#jwt-claim-verification[JSON web token claim verification] section.
 It applies to ID tokens but also to access tokens in a JWT format if the `web-app` application has requested the access token verification.
```
```suggestion
It applies to ID and access tokens in a JWT format if the `web-app` application has requested the access token verification.
```

sheilamjones replied:
@MichalMaler - Modifying this further slightly as it could be confusing an imply that both the "ID tokens" and the "access tokens" are in a JWT format. My understanding of the original sentence is that only the "access tokens" are in a JWT format.
Proposing instead to change to the following: "It applies to ID tokens and **also to* access tokens in a JWT format, if the `web-app` application has requested the access token verification."

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -552,18 +568,18 @@ link:https://datatracker.ietf.org/doc/html/rfc7636[Proof Key for Code Exchange]
 While PKCE is of primary importance to public OpenID Connect clients, such as SPA scripts running in a browser, it can also provide an extra level of protection to Quarkus OIDC `web-app` applications.
```
```suggestion
While PKCE is of primary importance to public OpenID Connect clients, such as SPA scripts running in a browser, it can also provide extra protection to Quarkus OIDC `web-app` applications.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -552,18 +568,18 @@ link:https://datatracker.ietf.org/doc/html/rfc7636[Proof Key for Code Exchange]
 While PKCE is of primary importance to public OpenID Connect clients, such as SPA scripts running in a browser, it can also provide an extra level of protection to Quarkus OIDC `web-app` applications.
 With PKCE, Quarkus OIDC `web-app` applications are confidential OpenID Connect clients capable of securely storing the client secret and using it to exchange the code for the tokens.
```
```suggestion
With PKCE, Quarkus OIDC `web-app` applications act as confidential OpenID Connect clients.
They securely store the client secret and exchange the code for tokens.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -552,18 +568,18 @@ link:https://datatracker.ietf.org/doc/html/rfc7636[Proof Key for Code Exchange]
 
-If you already have a 32-characters client secret then you do not need to set the  `quarkus.oidc.authentication.pkce-secret` property unless you prefer to use a different secret key.
-This secret will be autogenerated if it is not configured and if the fallback to the client secret is not possible in case of the client secret being less than 16 characters long.
+If you already have a 32-characters client secret, then you do not need to set the  `quarkus.oidc.authentication.pkce-secret` property unless you prefer to use a different secret key.
```
```suggestion
If you already have a 32-character client secret, you do not need to set the  `quarkus.oidc.authentication.pkce-secret` property unless you prefer a different secret key.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -552,18 +568,18 @@ link:https://datatracker.ietf.org/doc/html/rfc7636[Proof Key for Code Exchange]
+This secret will be auto-generated if it is not configured and if the fallback to the client secret is not possible in cases where the client secret is less than 16 characters long.
 
-The secret key is required for encrypting a randomly generated `PKCE` `code_verifier` while the user is being redirected with the `code_challenge` query parameter to an OIDC provider to authenticate.
+The secret key is required for encrypting a randomly generated PKCE `code_verifier` while the user is being redirected with the `code_challenge` query parameter to an OIDC provider to authenticate.
```
```suggestion
The secret key is required to encrypt a randomly generated PKCE `code_verifier` while the user is redirected with the `code_challenge` query parameter to an OIDC provider to authenticate.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -580,7 +596,8 @@ Use the following key points to find the right balance for securing your Quarkus
 This state is a key element controlling the lifetime of authentication data.
 
-Use the `quarkus.oidc.authentication.cookie-path` property to ensure that the same cookie is visible when you access protected resources with overlapping or different roots, for example:
+Use the `quarkus.oidc.authentication.cookie-path` property to ensure that the same cookie is visible when you access protected resources with overlapping or different roots.
```
```suggestion
Use the `quarkus.oidc.authentication.cookie-path` property to ensure the same cookie is visible when accessing protected resources with overlapping or different roots.

```

sheilamjones replied:
Propose to keep this one as is for clarity on who is accessing the protected resources.

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -953,14 +976,15 @@ quarkus.oidc.application-type=web-app
 ----
 
-This path will be compared against the current request's path and the user will be logged out if these paths match.
+This path will be compared to the current request's path and the user will be logged out if these paths match.
```
```suggestion
This path will be compared to the current request's path, and the user will be logged out if these paths match.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -996,16 +1020,17 @@ More useful methods will be added to it over time.
 
-By default, logout is based on the expiration time of the ID token issued by the OIDC provider. When the ID token expires, the current user session at the Quarkus endpoint is invalidated and the user is redirected to the OIDC provider again to authenticate.
+By default, logout is based on the expiration time of the ID token issued by the OIDC provider.
+When the ID token expires, the current user session at the Quarkus endpoint is invalidated and the user is redirected to the OIDC provider again to authenticate.
```
```suggestion
When the ID token expires, the current user session at the Quarkus endpoint is invalidated, and the user is redirected to the OIDC provider again to authenticate.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -996,16 +1020,17 @@ More useful methods will be added to it over time.
+If you have a xref:security-oidc-bearer-token-authentication.adoc#single-page-applications[single page application for service applications] where your OIDC provider script such as `keycloak.js` is managing an authorization code flow, then that script will also control the SPA authentication session lifespan.
+====
 
 If you work with a Quarkus OIDC `web-app` application, then it is the Quarkus OIDC code authentication mechanism that is managing the user session lifespan.
```
```suggestion
If you work with a Quarkus OIDC `web-app` application, then the Quarkus OIDC code authentication mechanism manages the user session lifespan.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1014,10 +1039,10 @@ The session age should be longer than the ID token lifespan and close to or equa
---
+====
 You use only the `quarkus.oidc.authentication.session-age-extension` property to significantly extend the session lifespan, if required.
 You use the `quarkus.oidc.token.lifespan-grace` property only for taking some small clock skews into consideration.
```
```suggestion
You use the `quarkus.oidc.token.lifespan-grace` property only to consider some small clock skews.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1027,54 +1052,55 @@ This refresh token might also be recycled (refreshed) itself as part of this pro
 As a result, the new session cookie is created and the session is extended.
```
```suggestion
As a result, the new session cookie is created, and the session is extended.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1027,54 +1052,55 @@ This refresh token might also be recycled (refreshed) itself as part of this pro
 
 You can go one step further and proactively refresh ID tokens or access tokens that are about to expire.
-Set `quarkus.oidc.token.refresh-token-time-skew` to the value you want to anticipate the refresh. If, during the current user request, it is calculated that the current ID token will expire within this `quarkus.oidc.token.refresh-token-time-skew` then it will be refreshed and the new session cookie will be created. This property should be set to a value which is less than the ID token lifespan; t ...
+Set `quarkus.oidc.token.refresh-token-time-skew` to the value you want to anticipate the refresh. If, during the current user request, it is calculated that the current ID token will expire within this `quarkus.oidc.token.refresh-token-time-skew`, then it will be refreshed and the new session cookie will be created. This property should be set to a value which is less than the ID token lifespan;  ...
```
```suggestion
Set `quarkus.oidc.token.refresh-token-time-skew` to the value you want to anticipate the refresh. If, during the current user request, it is calculated that the current ID token will expire within this `quarkus.oidc.token.refresh-token-time-skew`, then it will be refreshed, and the new session cookie will be created.
This property should be set to a value less than the ID token lifespan; the closer it is to this lifespan value, the more often the ID token will be refreshed.
```

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1027,54 +1052,55 @@ This refresh token might also be recycled (refreshed) itself as part of this pro
-Set `quarkus.oidc.token.refresh-token-time-skew` to the value you want to anticipate the refresh. If, during the current user request, it is calculated that the current ID token will expire within this `quarkus.oidc.token.refresh-token-time-skew` then it will be refreshed and the new session cookie will be created. This property should be set to a value which is less than the ID token lifespan; t ...
+Set `quarkus.oidc.token.refresh-token-time-skew` to the value you want to anticipate the refresh. If, during the current user request, it is calculated that the current ID token will expire within this `quarkus.oidc.token.refresh-token-time-skew`, then it will be refreshed and the new session cookie will be created. This property should be set to a value which is less than the ID token lifespan;  ...
 
 You can further optimize this process by having a simple JavaScript function ping your Quarkus endpoint to periodically emulate the user activity, which minimizes the time frame during which the user might have to be re-authenticated.
```
```suggestion
You can further optimize this process by having a simple JavaScript function ping your Quarkus endpoint periodically to emulate the user activity, which minimizes the time frame during which the user might have to be re-authenticated.
```

sheilamjones commented:
Update: setting back to Draft until all updates are done and approvals are received.

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1146,7 +1172,8 @@ public class TokenResource {
 ----
 
-If you support more than one social provider with the help of xref:security-openid-connect-multitenancy.adoc[OpenID Connect Multi-Tenancy], for example, Google which is an OpenID Connect Provider returning `IdToken` and GitHub which is an OAuth2 provider returning no `IdToken` and only allowing to access `UserInfo` then you can have your endpoint working with only the injected `SecurityIdentity`  ...
+If you support more than one social provider with the help of xref:security-openid-connect-multitenancy.adoc[OpenID Connect Multi-Tenancy], for example, Google which is an OIDC provider that returns `IdToken` and GitHub which is an OAuth2 provider that does not return `IdToken` and only allows access to `UserInfo`, then you can have your endpoint working with only the injected `SecurityIdentity`  ...
```
```suggestion
If you support more than one social provider with the help of xref:security-openid-connect-multitenancy.adoc[OpenID Connect Multi-Tenancy], for example, Google, which is an OIDC provider that returns `IdToken`, and GitHub, which is an OAuth2 provider that does not return `IdToken` and only allows access to `UserInfo`, then you can have your endpoint working with only the injected `SecurityIdentity` for both Google and GitHub flows.
```

sheilamjones replied:
Thank you. Good catch!

MichalMaler commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1333,7 +1360,8 @@ For more information, see the xref:security-cors.adoc#cors-filter[CORS filter] s
 
 You can have Quarkus OIDC `web-app` applications access **Google Cloud services** such as **BigQuery** on behalf of the currently-authenticated users who have enabled OpenID Connect (Authorization Code Flow) permissions to such services in their Google developer consoles.
```
```suggestion
You can have Quarkus OIDC `web-app` applications access **Google Cloud services** such as **BigQuery** on behalf of the currently authenticated users who have enabled OpenID Connect (Authorization Code Flow) permissions to such services in their Google developer consoles.
```

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -294,7 +298,7 @@ quarkus.oidc.introspection-credentials.secret=introspection-user-secret
 ==== OIDC request filters
 
-You can filter OIDC requests made by Quarkus to the OIDC provider by registering one or more `OidcRequestFiler` implementations, which can update or add new request headers, as well as log requests.
+You can filter OIDC requests made by Quarkus to the OIDC provider by registering one or more `OidcRequestFiler` implementations, which can update or add new request headers and log requests.
```
`and` makes logging somehow related to the process of adding new headers, but it can be an independent activity

sheilamjones replied:
modified slightly to say "and can also log requests."

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -354,26 +358,26 @@ public class OidcDiscoveryRequestCustomizer implements OidcRequestFilter {
     }
 }
 ----
-<1> Restrict this filter to requests targeting the OIDC discovery endpoint only.
```
IMHO it does not harm to clarify where exactly this enum is used

sheilamjones replied:
Hi @sberyozkin, in your previous review of this topic before PTO, you proposed to remove this line 357 and I removed it and the corresponding callout number  "<1>". I'm not sure what to update there. Would you like me to re-insert this line or a modified version of it perhaps?

sberyozkin replied:
@sheilamjones Oh, looks like I keep changing my mind :-), sorry. Yeah, it looks like it can be a useful piece of info, please put it back

sheilamjones replied:
No worries. Thanks @sberyozkin. Re-inserted this line.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -522,48 +537,49 @@ This is done by ensuring tokens can be trusted.
+There will be one or two remote calls with the code flow access token if the token introspection, `UserInfo`, or both are required.
 
-For more information about using a default token cache or registering a custom cache implementation, see xref:security-oidc-bearer-token-authentication.adoc#token-introspection-userinfo-cache[Token Introspection and UserInfo cache].
+For more information about using a default token cache or registering a custom cache implementation, see xref:security-oidc-bearer-token-authentication.adoc#token-introspection-userinfo-cache[Token introspection and UserInfo cache].
```
Should it be `the default token cache` ? I see `a` was already there but since it is a single, well known cache then may be `the` works

sheilamjones replied:
Yes, I think "the default cache" also works here. Updated.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -522,48 +537,49 @@ This is done by ensuring tokens can be trusted.
 
-For information about the claim verification, including the `iss` (issuer) claim, see the xref:security-oidc-bearer-token-authentication.adoc#jwt-claim-verification[JSON Web Token Claim verification] section.
-It applies to ID tokens but also to access tokens in a JWT format if the `web-app` application has requested the access token verification.
+For information about the claim verification, including the `iss` (issuer) claim, see the xref:security-oidc-bearer-token-authentication.adoc#jwt-claim-verification[JSON web token claim verification] section.
```
```suggestion
For information about the claim verification, including the `iss` (issuer) claim, see the xref:security-oidc-bearer-token-authentication.adoc#jwt-claim-verification[JSON Web Token claim verification] section.
```

sheilamjones replied:
oops, missed this one. Corrected. Thank you.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -663,10 +683,10 @@ You can disable token encryption in the session cookie by setting `quarkus.oidc.
 
 For example, you might want to keep the tokens in a cache cluster and have only a key stored in a session cookie.
-Note that this approach might introduce some challenges if you need to make the tokens available across multiple microservices nodes.
+If you need to make the tokens available across multiple microservices nodes, not that this approach might introduce some challenges.
```
The removed line reads quite well IMHO, is it not quite correct ? Minor typo in `not` but I'd rather keep the original sentence if possible

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -996,85 +1020,91 @@ More useful methods will be added to it over time.
 
-OAuth2 providers such as GitHub do not return `IdToken`, the fact of the user authentication is implicit and is indirectly represented by the `access` token which represents an authenticated user authorizing the current Quarkus `web-app` application to access some data on behalf of the authenticated user.
+OAuth2 providers such as GitHub do not return `IdToken`.
+The user authentication is implicit and is indirectly represented by the `access` token, which represents an authenticated user authorizing the current Quarkus `web-app` application to access some data on behalf of the authenticated user.
```
It makes it somehow disconnected from the previous one, as it is not obvious it applies only to that one or if it is a more general statement. IMHO, having a comma is OK, as far as I recall the idea of a sentence per new line was about separate sentences only

sheilamjones replied:
Hi @sberyozkin, You're right about having separate sentences as a new sentence per line . In this case, the original sentence is very long, which throws a warning in Vale. I take your point though about possible disconnect between the sentence. What about the following proposal instead, i.e. splitting the sentence before the "which..."
"OAuth2 providers such as GitHub do not return `IdToken`, and the user authentication is implicit and indirectly represented by the `access` token.
**The `access` token represents** an authenticated user authorizing the current Quarkus `web-app` application to access some data on behalf of the authenticated user."

sheilamjones replied:
Done.

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1364,13 +1397,13 @@ quarkus.oidc.token.issuer=https://accounts.google.com
 === Running Quarkus application behind a reverse proxy
 
-OIDC authentication mechanism can be affected if your Quarkus application is running behind a reverse proxy, gateway, or firewall when HTTP `Host` header might be reset to the internal IP address, HTTPS connection might be terminated, and so on.
+An OIDC authentication mechanism can be affected if your Quarkus application is running behind a reverse proxy, gateway, or firewall when HTTP `Host` header might be reset to the internal IP address and HTTPS connection might be terminated, and so on.
```
Should it be `The` as it is one of the well known Quarkus security mechanisms ?

sberyozkin commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1601,9 +1640,9 @@ public class CodeFlowAuthorizationTest {
+If you need to do the integration testing against Keycloak then you are encouraged to do it with <<integration-testing-keycloak-devservices,Dev Services for Keycloak>>.
 
-Start with adding the following dependency:
+Start by adding the following dependency:
```
The bearer token guide uses `Start with`, `Start by` is fine if it is better, worth making it consistent IMHO across both docs

sheilamjones replied:
We modified the tutorial guide slightly to say "First, add the following dependency:" I will make that same update here for consistency.

sberyozkin commented:
Hey @sheilamjones, a few more suggestions here, a nice cleanup over all, thanks

sheilamjones commented on docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc:
```diff
@@ -1657,15 +1694,17 @@ public class CodeFlowAuthorizationTest {
 === TestSecurity annotation
 
-See the xref:security-oidc-bearer-token-authentication.adoc#integration-testing-security-annotation[Use TestingSecurity with injected JsonWebToken] section for more information about using `@TestSecurity` and `@OidcSecurity` annotations for testing the `web-app` application endpoint code, which depends on the injected ID and access `JsonWebToken` as well as `UserInfo` and `OidcConfigurationMetada ...
+For more information about using `@TestSecurity` and `@OidcSecurity` annotations for testing the `web-app` application endpoint code, which depends on the injected ID and access `JsonWebToken`, `UserInfo`, and `OidcConfigurationMetadata`, see xref:security-oidc-bearer-token-authentication.adoc#integration-testing-security-annotation[Use TestingSecurity with injected JsonWebToken].
```
Hi @sberyozkin 

The Bearer tutorial topic has the following:
"You can use `@TestSecurity` and `@OidcSecurity` annotations to test the `service` application endpoint code that depends on the injected `JsonWebToken`, `UserInfo`, `OidcConfigurationMetadata`, or all three.
I propose to update this topic to the above statement and to also add:
"For more information, see xref:security-oidc-bearer-token-authentication.adoc#integration-testing-security-annotation[Use TestingSecurity with injected JsonWebToken]."

Would this work?

sheilamjones replied:
Hi @sberyozkin,
I updated this text to the following. Would this work? If okay, I can also update the Bearer token tutorial topic to the same.
"You can use `@TestSecurity` and `@OidcSecurity` annotations to test the  `web-app` application endpoint code, which depends on either one of the following injections, or all four:
- ID `JsonWebToken`
- Access `JsonWebToken
- `UserInfo`
- `OidcConfigurationMetadata`"

sberyozkin reviewed (approved):
Great work @sheilamjones, thanks for your patience.

Merging now
